pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1974,6 +1975,22 @@ private Set<String> findConsumingSegments(IdealState idealState) {
         }
       }
     });
+    // For pauseless tables, a segment marked ONLINE in the ideal state may not have been committed yet.
+    // We rely on SegmentZkMetadata to determine whether a segment has been committed (status is DONE)
+    // instead of relying solely on the ideal state.
+    // A segment in COMMITTING state is treated as consuming for pauseStatus.
+    String tableNameWithType = idealState.getResourceName();
+    if (PauselessConsumptionUtils.isPauselessEnabled(getTableConfig(tableNameWithType))) {
+      Map<Integer, SegmentZKMetadata> metadataMap = getLatestSegmentZKMetadataMap(tableNameWithType);
```

Comment:
I feel this fetch can be too heavy if there are a lot of segments in the table
would it be better if we first fetched EV
and then fetched metadata only for segments that are ONLINE in IS but CONSUMING in EV?